### PR TITLE
Connections console: Add Angular badge for Angular plugins

### PR DESCRIPTION
**Pull request by xnyo**
**What is this feature?**

Adds "Angular" badge next to Angular plugins to the on-prem connections console.
image

**Why do we need this feature?**

Angular deprecation

**Who is this feature for?**

Grafana users who rely on Angular datasources

**Which issue(s) does this PR fix?**:

Related to  #68974

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/connections/tabs/ConnectData/CardGrid/CardGrid.tsx [hunk 1]
```diff
@@ -39,7 +40,14 @@ const getStyles = (theme: GrafanaTheme2) => ({
...
 });
 
-export type CardGridItem = { id: string; name: string; description: string; url: string; logo?: string };
+export type CardGridItem = {
+  id: string;
+  name: string;
+  description: string;
+  url: string;
+  logo?: string;
+  angularDetected?: boolean;
```
Comment (xnyo) [inline public/app/features/connections/tabs/ConnectData/CardGrid/CardGrid.tsx:49]: I have only added this property, the rest is the same

File: public/app/features/connections/tabs/ConnectData/CardGrid/CardGrid.tsx [hunk 2]
```diff
@@ -68,7 +76,10 @@ export const CardGrid = ({ items, onClickItem }: CardGridProps) => {
                   <img src={item.logo} alt={`logo of ${item.name}`} />
                 </div>
               )}
-              <h4 className={styles.label}>{item.name}</h4>
+              <div>
+                <h4 className={styles.label}>{item.name}</h4>
+                {item.angularDetected && <PluginAngularBadge />}
```
Comment (academo) [inline public/app/features/connections/tabs/ConnectData/CardGrid/CardGrid.tsx:81]: Prefer `{item.angularDetected? <PluginAngularBadge />: null}`

**Review by academo — COMMENTED** [review]
Left a comment about the condition. Not only a matter of preference but safer code. see https://codesandbox.io/s/autumn-sunset-dp7vdj?file=/src/App.js

File: public/app/features/connections/tabs/ConnectData/CardGrid/CardGrid.tsx (hunk 2 above)
Comment (xnyo) [reply public/app/features/connections/tabs/ConnectData/CardGrid/CardGrid.tsx]: Thanks :pray: ! done

**Review by academo — APPROVED** [review]

**Review by sympatheticmoose — APPROVED** [review]
LGTM

**Comment by torkelo** [comment]
The design of these cards (and placement of the badges) look a bit unpolished. Say compared to plugin list:

Screenshot 2023-07-07 at 14 20 24

**Comment by torkelo** [comment]
I pushed a branch here with some changes to how to correctly extend/modify Card component to add things like badges, I also aligned the styling with the data source cards (used in new data source page), cards that looks so much better. 

Screenshot 2023-07-07 at 15 03 15

I think with description and the [Core] Badge (or Installed) to also mark which data sources are bundled plugins (or installed) would also be useful. And this list really needs some default sorting to show show the popular ones first (or popular in a separate group). 

With description (did not add the [Core] or [Installed] badges, as it requires more changes to the component props to pass in that info 
Screenshot 2023-07-07 at 15 04 27

**Comment by torkelo** [comment]
https://github.com/grafana/grafana/compare/connections-card-design

**Comment by xnyo** [comment]
Thank you,  I will merge it with this branch :raised_hands:

**Comment by xnyo** [comment]
I have merged those changes into this PR's code. I decided to keep the description hidden for now, since it messes up the spacing if there's too much text, and the purpose of this PR was just to add the Angular badge. e.g.:

image

The other changes can be done in another PR if that's okay, I sent the feedback to connections working group, thanks :)

**Comment by torkelo** [comment]
@xnyo yea, let's skip description for now. Needs some fine tuning to skip / cut long descriptions

**Comment by torkelo** [comment]
And the card grid template needs update, it positions the description strangly

**Review by mckn — APPROVED** [review]
LGTM! Great suggestions by @torkelo

**Comment by mikkancso** [comment]
About showing the description on the cards:
We already have a plan to improve how these cards look in the catalog. Please feel free to submit any feedback in [this thread](https://raintank-corp.slack.com/archives/C034LMXLXQU/p1688760676526659). 
We are planning to work on this in Q3, so please don't do ad-hoc improvements without consulting us, because it will only make our job more difficult.

**Pull request merged**